llarp/rpc/rpc_server.cpp

```diff
@@ -107,626 +61,527 @@ namespace llarp::rpc
+  void
+  RPCServer::invoke(SwapExits& swapexits)
+  {
+    MapExit map_request;
+    UnmapExit unmap_request;
+    auto endpoint = m_Router.hiddenServiceContext().GetDefault();
+    auto current_exits = endpoint->ExtractStatus()["exitMap"];
+
+    if (current_exits.empty())
+    {
+      SetJSONError("Cannot swap to new exit: no exits currently mapped", swapexits.response);
+      return;
+    }
+
+    // steal replier from swapexit RPC endpoint
+    unmap_request.replier.emplace(swapexits.move());
+
+    // set map_exit request to new address
+    map_request.request.address = swapexits.request.exit_addresses[1];
+
+    // set token for new exit node mapping
+    if (not swapexits.request.token.empty())
+      map_request.request.token = swapexits.request.token;
+
+    // populate map_exit request with old IP ranges
+    for (auto& [range, exit] : current_exits.items())
+    {
+      if (exit.get<std::string>() == swapexits.request.exit_addresses[0])
+      {
+        map_request.request.ip_range.emplace_back(range);
+        unmap_request.request.ip_range.emplace_back(range);
+      }
+    }
+
+    if (map_request.request.ip_range.empty() or unmap_request.request.ip_range.empty())
+    {
+      SetJSONError("No mapped ranges found matching requested swap", swapexits.response);
+      return;
+    }
+
+    endpoint->map_exit(
+        map_request.request.address,
+        map_request.request.token,
+        map_request.request.ip_range,
+        [unmap = std::move(unmap_request),
+         &r = m_Router,
+         ep = endpoint,
+         old_exit = swapexits.request.exit_addresses[0]](bool success, std::string result) mutable {
+          if (not success)
+            unmap.send_response({{"error"}, std::move(result)});
+          else
+          {
+            try
             {
-              const auto itr = obj.find("override");
-              if (itr != obj.end())
-              {
-                if (not itr->is_object())
-                {
-                  reply(CreateJSONError("override is not an object"));
-                  return;
-                }
-                for (const auto& [section, value] : itr->items())
-                {
-                  if (not value.is_object())
-                  {
-                    reply(CreateJSONError(
-                        fmt::format("failed to set [{}]: section is not an object", section)));
-                    return;
-                  }
-                  for (const auto& [key, value] : value.items())
-                  {
-                    if (not value.is_string())
-                    {
-                      reply(CreateJSONError(fmt::format(
-                          "failed to set [{}]:{}: value is not a string", section, key)));
-                      return;
-                    }
-                    r->GetConfig()->Override(section, key, value.get<std::string>());
-                  }
-                }
-              }
+              r.routePoker()->Down();
```

Comment:
this will 100% cause packets to leak out, which is what we want to prevent with this endpoint.